ts/components/services/ServicesSearch.tsx

```diff
@@ -36,13 +40,13 @@ type State = {
  */
 const generateSectionsServicesStateMatchingSearchTextArrayAsync = (
   // tslint:disable-next-line: readonly-array
-  servicesState: Array<SectionListData<pot.Pot<ServicePublic, Error>>>,
+  servicesState: ReadonlyArray<ServicesSectionState>,
   searchText: string
   // tslint:disable-next-line: readonly-array
-): Promise<Array<SectionListData<pot.Pot<ServicePublic, Error>>>> => {
+): Promise<ReadonlyArray<ServicesSectionState>> => {
   return new Promise(resolve => {
     // tslint:disable-next-line: readonly-array
-    const result: Array<SectionListData<pot.Pot<ServicePublic, Error>>> = [];
+    const result: ServicesSectionState[] = [];
```

Comment:
```suggestion
    return new Promise(resolve => {
    const result = servicesState.filter(sectionList => {
      const filtered = filterSectionListDataMatchingSearchText(
        sectionList,
        searchText
      );
      return filtered != null;
    });
    resolve(result);
  });
```

you could use **filter** instead of **foreach**

Reply:
updated within https://github.com/teamdigitale/io-app/pull/1338/commits/4f6eb900a7db5adb0a7a43ea2af861eec0bb6c9b